agent/src/apiclient.rs

```diff
@@ -0,0 +1,353 @@
+/*!
+apiclient is a client for interacting with the Bottlerocket Update API.
+Brupop volume mounts apiclient binary into the agent container.
+
+Bottlerocket Update API: https://github.com/bottlerocket-os/bottlerocket/tree/develop/sources/updater
+Bottlerocket apiclient: https://github.com/bottlerocket-os/bottlerocket/tree/develop/sources/api/apiclient
+*/
+
+use serde::Deserialize;
+use snafu::{ensure, ResultExt, Snafu};
+use std::process::{Command, Output};
+use tokio::time::{sleep, Duration};
+
+const API_CLIENT_BIN: &str = "apiclient";
+const UPDATES_STATUS_URI: &str = "/updates/status";
+const OS_URI: &str = "/os";
+const REFRESH_UPDATES_URI: &str = "/actions/refresh-updates";
+const PREPARE_UPDATES_URI: &str = "/actions/prepare-update";
+const ACTIVATE_UPDATES_URI: &str = "/actions/activate-update";
+const REBOOT_URI: &str = "/actions/reboot";
+const MAX_ATTEMPTS: i8 = 5;
+const UPDATE_API_SLEEP_DURATION: Duration = Duration::from_millis(10000);
+const UPDATE_API_BUSY_STATUSCODE: &str = "423";
+
+/// The module-wide result type.
+pub type Result<T> = std::result::Result<T, ApiClientError>;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility = "pub")]
+
+pub enum ApiClientError {
```

Comment:
Maybe a painful refactor, but consider moving errors at the end, and follow the "news paper" style, where the most important parts of your PR are at the top. Also consider prefixing the errors with `error`:

```suggestion
mod error {
    pub(super) enum Error {
```

And then you can use the errors as: `error::Update`

But before all that, why aren't these errors part of the `error.rs` file?